doc/sphinx/proofs/writing-proofs/rewriting.rst

```diff
@@ -90,11 +93,63 @@ Rewriting with Leibniz and setoid equality
      any of them can be rewritten.  If not specified, only the first occurrence
      in the conclusion is replaced.
 
-     If :n:`at @occs_nums` is specified, rewriting is always done with
-     :ref:`setoid rewriting <generalizedrewriting>`, even for Leibniz’s equality,
-     which means that you must `Require Setoid` to use that form.
-     However, note that :tacn:`rewrite` (even when using setoid rewriting) and
-     :tacn:`setoid_rewrite` don't behave identically (as already mentioned above).
+     .. note::
+
+        If :n:`at @occs_nums` is specified, rewriting is always done
+        with :ref:`setoid rewriting <generalizedrewriting>`, even for
+        Leibniz equality, which means that you must `Require
+        Setoid` to use that form.  However, note that :tacn:`rewrite`
+        (even when using setoid rewriting) and :tacn:`setoid_rewrite`
+        don't behave identically (as already mentioned above).
+
+     .. note::
+
+        For each selected hypothesis (and the conclusion),
+        :tacn:`rewrite` will instantiate the rewriting lemma first
+        before taking the `at` clause into account, which limits the
+        number of occurrences that can be rewritten.  You can force
+        :tacn:`rewrite` to instantiate the rewriting lemma differently
+        by using a `with` clause (see :n:`@one_term_with_bindings`) or
+        you can use :tacn:`setoid_rewrite`, which considers each
+        occurrence separately.
+
+        .. example::
+
+           .. coqtop:: none
+
+              Require Import Arith.
+
+           If we want to rewrite the right-hand side, it will not work
+           like this:
+
+           .. coqtop:: out
+
+              Lemma add_assoc x y : x + y = y + x.
```

Comment:
Would you modify this so it is indeed the associative property?  Here and below.  (misnamed and perhaps confusingly equivalent to `Nat.add_comm`)

Reply:
Ah right, this was my initial example. I forgot to change the name.